src/bgem/bspline/bspline_approx.py

```diff
@@ -379,6 +383,32 @@ def compute_default_nuv(self):
             raise Exception("Two few points, {}, to make approximation, degree: {}".format(self._n_points, self._degree))
         return self.nuv
 
+    def _refine_knots(self, knot, ref_vec):
+
+        knotlist = []
+        for i in range(0, len(knot) - 5):
```

Comment:
Replace 5 and 3 with deg to make the code more general and descriptive even if we have degree fixed right now.